sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -210,20 +218,28 @@ private synchronized boolean readNextRecord() throws IOException {
           return false;
         }
 
-        // N.B.: For simplicity, we update fractionConsumed once a new response is fetched, not
-        // when we reach the end of the current response. In practice, this choice is not
-        // consequential.
-        fractionConsumed = fractionConsumedFromLastResponse;
+        fractionConsumedFromPreviousResponse = fractionConsumedFromCurrentResponse;
         ReadRowsResponse nextResponse = responseIterator.next();
         decoder =
             DecoderFactory.get()
                 .binaryDecoder(
                     nextResponse.getAvroRows().getSerializedBinaryRows().toByteArray(), decoder);
-        fractionConsumedFromLastResponse = getFractionConsumed(nextResponse);
+        rowsReadFromCurrentResponse = 0L;
+        rowCountFromCurrentResponse = nextResponse.getAvroRows().getRowCount();
+        fractionConsumedFromCurrentResponse = getFractionConsumed(nextResponse);
```

Comment:
I think we need to adjust variable names here. Current naming makes it pretty confusing. Should this "nextResponse" be named "currentResponse" since it result in "fractionConsumedFromCurrentResponse" ?

Reply:
Done.